tests/test_downloads.py

```
@@ -0,0 +1,139 @@
+    def test_actual_download(self, httpbin):
+        url = httpbin.url + '/robots.txt'
+        body = urlopen(url).read().decode()
+        env = TestEnvironment(stdin_isatty=True, stdout_isatty=False)
+        r = http('--download', url, env=env)
+        assert 'Downloading' in r.stderr
+        assert '[K' in r.stderr
+        assert 'Done' in r.stderr
+        assert body == r
+
+    def test_download_with_Content_Length(self, httpbin):
+        devnull = open(os.devnull, 'w')
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)